src/mode/modeHandler.ts

```diff
@@ -650,9 +650,17 @@ export class ModeHandler implements vscode.Disposable {
       this.vimState.currentMode !== ModeName.VisualBlock
     ) {
       // Number of selections changed, make sure we know about all of them still
-      this.vimState.allCursors = e.textEditor.selections.map(
-        x => new Range(Position.FromVSCodePosition(x.start), Position.FromVSCodePosition(x.end))
-      );
+      let cursors: Range[];
+      cursors = [];
+      for (let sel of e.textEditor.selections) {
```

Comment:
@westim Any reason why you chose `for ... of` instead of the `map`

Reply:
I wasn't sure if it was still possible to capture the `if` statement in a call to `map`, and writing it out this way made the logic more clear to me. Otherwise, there's no semantic justification for using `for...of`.

Reply:
I would prefer to keep the `map` and just change the logic inside